plugin/storage/es/spanstore/reader.go

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package spanstore
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/olivere/elastic"
+	"go.uber.org/zap"
+
+	"github.com/uber/jaeger/model"
+	"github.com/uber/jaeger/pkg/es"
+	"github.com/uber/jaeger/storage/spanstore"
+)
+
+// SpanReader can query for and load traces from ElasticSearch
+type SpanReader struct {
+	ctx    context.Context
+	client es.Client
+	logger *zap.Logger
+}
+
+// NewSpanReader returns a new SpanReader.
+func NewSpanReader(client es.Client, logger *zap.Logger) *SpanReader {
+	ctx := context.Background()
+	return &SpanReader{
+		ctx:    ctx,
+		client: client,
+		logger: logger,
+	}
+}
+
+// GetTrace takes a traceID and returns a Trace associated with that traceID
+func (s *SpanReader) GetTrace(traceID model.TraceID) (*model.Trace, error) {
+	// TODO
+	return nil, nil
+}
+
+// Returns the array of indices that we need to query, based on query params
+func (s *SpanReader) findIndices(traceQuery spanstore.TraceQueryParameters) []string {
+	today := time.Now()
+	threeDaysAgo := today.AddDate(0, 0, -3)
+
+	if traceQuery.StartTimeMax.IsZero() || traceQuery.StartTimeMin.IsZero() {
+		traceQuery.StartTimeMax = time.Now()
+		traceQuery.StartTimeMin = time.Now().AddDate(0, 0, -3)
+	}
+
+	var indices []string
+	current := traceQuery.StartTimeMax
+	for current.After(traceQuery.StartTimeMin) && current.After(threeDaysAgo) {
+		index := "jaeger-" + current.Format("2006-01-02")
+		exists, _ := s.client.IndexExists(index).Do(s.ctx)
```

Comment:
if there's an error, `exists` is false so we don't care, we just don't add it to our list of indices

i'll add a comment in the next commit